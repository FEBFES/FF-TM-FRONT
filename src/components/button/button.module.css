.button {
  outline: none;
  font-size: 12px;
  padding: 20px 40px;
  width: fit-content;
  background: #0d0c22;
  border-radius: 8px;
  white-space: nowrap;
  -webkit-border-radius: 8px;
  -moz-border-radius: 8px;
  -ms-border-radius: 8px;
  -o-border-radius: 8px;
  color: aliceblue;
  transition: .6;
  -webkit-transition: .6;
  -moz-transition: .6;
  -ms-transition: .6;
  -o-transition: .6;
  border: none;
}

.fullWidth {
  width: 100%;
}

.button:hover {
  cursor: pointer;
  box-shadow: 0 0 4px 1px #9b9b9b;
}
.button:focus {
  outline: solid rgb(95, 143, 255); 
}