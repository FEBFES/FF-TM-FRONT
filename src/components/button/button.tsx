import React, { ButtonHTMLAttributes, forwardRef } from "react"
import styles from "./button.module.css"

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	title: string
	fullWidth?: boolean
}

const Button = forwardRef<HTMLButtonElement, ButtonProps>(({ title, fullWidth, ...props }, ref) => {
	return (
		<button
			{...props}
			ref={ref}
			className={`${styles.button} ${fullWidth && styles.fullWidth} ${props.className}`}>
			{title}
		</button>
	)
})

export default Button
