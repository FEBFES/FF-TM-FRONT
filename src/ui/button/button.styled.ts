import styled, { css, RuleSet } from 'styled-components';
import { IButtonVariant } from './button';

export const SButtonCloseType = styled.button`
  background: transparent;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 100px;
  outline: none;
  /*border: none;*/
  padding: 10px;
  width: 24px;
  height: 24px;
  font-size: 10px;
  color: var(--font-gray);
  border: 1px solid var(--font-gray);

  &:hover {
    cursor: pointer;
    color: var(--font-defautl);
    border: 1px solid var(--font-defautl);
  }
`;

export type StyledVariants<E extends string> = {
  [key in E]?: RuleSet;
};

const buttonsVariants: StyledVariants<IButtonVariant> = {
  'submit': css`
    background: #168d1b;
    color: #ffffff;

    &:hover {
      background-color: #20b427;
    }
  `,
  'danger': css`
    background: #fc4143;
    color: white;

    &:hover {
      background-color: #ff585a;
    }
  `,
  'primary': css`
    color: white;
    background: var(--bg-primary);

    &:hover {
      background-color: #1677ff;
    }
  `,
  'secondary': css`
    background: #606060;
    color: white;

    &:hover {
      background-color: #606060;
    }
  `,
};

export const SBasicButton = styled.button<{ variant: IButtonVariant }>`
  ${({ variant }) => buttonsVariants[variant]}
  margin: 5px;
  display: flex;
  align-items: center;
  padding: 5px 15px;
  border: none;
  border-radius: 6px;
  font-size: 14px;
  font-weight: 500;
  transition: 0.2s;
  max-width: fit-content;
  white-space: nowrap;

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.5;
  }
`;
