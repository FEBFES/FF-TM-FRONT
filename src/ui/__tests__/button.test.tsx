import React from 'react';
import { Button } from '../button/button';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
  it('SNAPSHOTS: UI - Button component - render', () => {
    const elem = renderer.create(<Button />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (Loaded: true) component - render', () => {
    const elem = renderer.create(<Button load={true} />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (variant - primary) component - render', () => {
    const elem = renderer
      .create(<Button variant="primary" load={true} />)
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (variant - submit) component - render', () => {
    const elem = renderer.create(<Button variant="submit" />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (variant - danger) component - render', () => {
    const elem = renderer.create(<Button variant="danger" />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (variant - secondary) component - render', () => {
    const elem = renderer.create(<Button variant="secondary" />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (btnType - primary) component - render', () => {
    const elem = renderer.create(<Button btnType="primary" />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Button (btnType - close) component - render', () => {
    const elem = renderer.create(<Button btnType="close" />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
