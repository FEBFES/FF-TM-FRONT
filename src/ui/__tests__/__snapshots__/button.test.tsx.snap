// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> SNAPSHOTS: UI - Button (Loaded: true) component - render 1`] = `
<button
  className="sc-dkjKgF fywBev"
  onClick={[Function]}
  variant="primary"
>
  <div
    className="sc-bcPKhP haSZKe"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zm0 416c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM48 304c26.5 0 48-21.5 48-48s-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48zm464-48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM142.9 437c18.7-18.7 18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zm0-294.2c18.7-18.7 18.7-49.1 0-67.9S93.7 56.2 75 75s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zM369.1 437c18.7 18.7 49.1 18.7 67.9 0s18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</button>
`;

exports[`<Button /> SNAPSHOTS: UI - Button (btnType - close) component - render 1`] = `
<button
  className="sc-grXZZQ hgKUPA"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-xmark fa-sm "
    data-icon="xmark"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`<Button /> SNAPSHOTS: UI - Button (btnType - primary) component - render 1`] = `
<button
  className="sc-dkjKgF fywBev"
  onClick={[Function]}
  variant="primary"
/>
`;

exports[`<Button /> SNAPSHOTS: UI - Button (variant - danger) component - render 1`] = `
<button
  className="sc-dkjKgF dRaTpt"
  onClick={[Function]}
  variant="danger"
/>
`;

exports[`<Button /> SNAPSHOTS: UI - Button (variant - primary) component - render 1`] = `
<button
  className="sc-dkjKgF fywBev"
  onClick={[Function]}
  variant="primary"
>
  <div
    className="sc-bcPKhP haSZKe"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zm0 416c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM48 304c26.5 0 48-21.5 48-48s-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48zm464-48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM142.9 437c18.7-18.7 18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zm0-294.2c18.7-18.7 18.7-49.1 0-67.9S93.7 56.2 75 75s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zM369.1 437c18.7 18.7 49.1 18.7 67.9 0s18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</button>
`;

exports[`<Button /> SNAPSHOTS: UI - Button (variant - secondary) component - render 1`] = `
<button
  className="sc-dkjKgF cUknzn"
  onClick={[Function]}
  variant="secondary"
/>
`;

exports[`<Button /> SNAPSHOTS: UI - Button (variant - submit) component - render 1`] = `
<button
  className="sc-dkjKgF juvyuC"
  onClick={[Function]}
  variant="submit"
/>
`;

exports[`<Button /> SNAPSHOTS: UI - Button component - render 1`] = `
<button
  className="sc-dkjKgF fywBev"
  onClick={[Function]}
  variant="primary"
/>
`;
